src/ports/nfts/component.ts

```diff
@@ -110,7 +119,7 @@ export function createNFTComponent<T extends { id: string }>(options: {
       nfts: T[]
     }>(query, variables)
 
-    return nfts.map(fromFragment)
+    return nfts.map((f) => fromFragment(f))
```

Comment:
Is the caller missing from this call? 
Mention this because it doesn't seem necessary to transform it into an arrow function if so

Reply:
in this case `nfts.map(fromFragment)`, it will send the `index` as second parameter and it doesn't match to what the `caller` is expected to be (number vs string I believe)